feat: Separate business status from module status (#1628)
kyma-project/lifecycle-manager: docs/technical-reference/api/manifest-cr.md

### **.status**

The Manifest CR status is set based on the following logic, managed by the manifest reconciler: 

> - If the Module defined in the Manifest CR is successfully applied and the deployed Module is up and running, the status of the Manifest CR is set to `Ready`.

Reviewer: ```suggestion
- If the module defined in the Manifest CR is successfully applied and the deployed module is up and running, the status of the Manifest CR is set to `Ready`.
```

> - While the Manifest is being applied and the Deployment is still starting, the status of the Manifest CR is set to `Processing`.

Reviewer: ```suggestion
- While the manifest is being applied and the Deployment is still starting, the status of the Manifest CR is set to `Processing`.
```

> - If the Deployment is unable to start (for example, due to an `ImagePullBackOff` error) or if the application of the Manifest fails, the status of the Manifest CR is set to `Error`.

Reviewer: ```suggestion
- If the Deployment is unable to start (for example, due to an `ImagePullBackOff` error) or if the application of the manifest fails, the status of the Manifest CR is set to `Error`.
```

> - If the Deployment is unable to start (for example, due to an `ImagePullBackOff` error) or if the application of the manifest fails, the status of the Manifest CR is set to `Error`.

Reviewer 2: ```suggestion
- If the Deployment cannot start (for example, due to an `ImagePullBackOff` error) or if the application of the manifest fails, the status of the Manifest CR is set to `Error`.
```

- If the Manifest CR is marked for deletion, the status of the Manifest CR is set to `Deleting`.

> This status provides a reliable way to track the state of the Manifest CR and the associated Module, offering insights into the deployment process and any potential issues, while being decoupled of the Module business logic.

Reviewer: ```suggestion
This status provides a reliable way to track the state of the Manifest CR and the associated module, offering insights into the deployment process and any potential issues, while being decoupled from the module business logic.
```

> This status provides a reliable way to track the state of the Manifest CR and the associated module, offering insights into the deployment process and any potential issues, while being decoupled from the module business logic.

Reviewer 2: ```suggestion
This status provides a reliable way to track the state of the Manifest CR and the associated module. It offers insights into the deployment process and any potential issues while being decoupled from the module's business logic.
```

### **.status**

> The status of the Manifest Custom Resource (CR) is a direct, unaltered reflection of the [declarative status](../../../internal/declarative/README.md#resource-tracking), which means it follows the tracking process implemented by the internal library. There isn't a custom API specifically designed for this status tracking. Instead, the existing mechanisms provided by the library are utilized to monitor and update the status of the Manifest CR.  

Reviewer 3: thb, the first part is hard to understand for me (generated by GPT?), and we will soon remove the declarative pattern, there is no need to align this declarative status. I suggest removing the first paragraph.

The Manifest CR status is set based on the following logic, managed by the manifest reconciler: 

- If the Module defined in the Manifest CR is successfully applied and the deployed Module is up and running, the status of the Manifest CR is set to `Ready`.
- While the Manifest is being applied and the Deployment is still starting, the status of the Manifest CR is set to `Processing`.
- If the Deployment is unable to start (for example, due to an `ImagePullBackOff` error) or if the application of the Manifest fails, the status of the Manifest CR is set to `Error`.

> - If the Manifest CR is marked for deletion, the status of the Manifest CR is set to `Deleted`.

Reviewer 3: We don't have `Deleted` state.

Author: Typo, `Deleting` it is
